### DISCUSSION TITLE
Compilers, platforms C++ standard

### DISCUSSION THREAD

**Discussion by davidhunter22** [GabrielDosReis/ipr#257] [Ideas]
It would be useful if the README.md had details on what compilers, platforms and C++ the library supports.
Looking at the main.yml you can see what that is doing but I'm not sure everyone would know to look there
The following is in the yml

g++-10
clang++12 - It would be great to build two versions one with the GCC libstdc++ and the other with the clang libc++ standard library.
MSVC

Ir would be nice to add clang-cl on Windows - So the clang compiler pretending to be MSVC. 

The CMakeLists.txt seems to set the standard to C++20. Although this raises the question of how much of C++20, for instance libc++ does not yet support <format>

**Comment by GabrielDosReis** [comment]
The aim is to target MSVC on Windows, GCC and Clang on linux.  clang-cl is a non-goal.

> The CMakeLists.txt seems to set the standard to C++20. Although this raises the question of how much of C++20, for instance libc++ does not yet support 

All of C++20.  This project is forward looking, as opposed to conventional libraries that want to maximum coverage.  So, if libc++ is lacking something, then at some point, we may need to disable it until support comes online.  Backward compatibility isn't a goal at this point.  I would like to be able to move as quickly as possible.

Ideally, all recently released compiler versions are targeted; the only reason the workflow uses GCC-10 is because that is what is available in the ubuntu-based CI/CD in github.  If GCC-12 shows up, I will move it there.  Same for other compilers.
